docs/scripts_md/DCMSUM.md

```diff
@@ -0,0 +1,226 @@
+# NAME
+
+DICOM::DCMSUM -- deals with DICOM summaries for archiving and other purposes
+
+# SYNOPSIS
+
+# DESCRIPTION
+
+Deals with DICOM summaries for archiving and other purposes.
+
+## Methods
+
+### new($dcm\_dir, $tmp\_dir)(constructor)
+
+Creates a new instance of this class.
+
+INPUT: DICOM directory, target location
+
+RETURNS: a DICOM::DCMSUM object
+
+### database($dbh, $meta, $update, $tarType, $tarLog, $DCMmd5, ...)
+
+Inserts or updates the tarchive tables.
+
+INPUT:
+  - $dbh              : database handle
+  - $meta             : name of the .meta file
+  - $update           : set to 1 to update tarchive entry, 0 otherwise
+  - $tarType          : tar type version
+  - $tarLog           : name of the .log file
+  - $DCMmd5           : DICOM MD5SUM
+  - $Archivemd5       : tarchive MD5SUM
+  - $Archive          : archive location
+  - $neurodbCenterName: center name
+
+RETURNS: 1 on success
+
+### read\_file($file)
+
+Reads the content of a file (typically .meta file in the tarchive).
+
+INPUT: the file to be read
+
+RETURNS: the content of the file
+
+### acquistion\_count()
+
+Figures out the total number of acquisitions.
+
+RETURNS: number of acquisitions
+
+### file\_count()
+
+Figures out the total number of files.
+
+RETURNS: number of files
+
+### dcm\_count()
+
+Figures out the total number of DICOM files.
+
+RETURNS: number of DICOM files, or exits if no DICOM file found.
+
+### acquisition\_AoH($self->{dcminfo})
+
+Creates an Array of Hashes (AoH) describing acquisition parameters for each
+file.
+
+INPUT: list of DICOM files
+
+RETURNS: array of hashes with acquisition parameters for each file
+
+### collapse($self->{acqu\_AoH})
+
+Collapses the AoH to get a summary of acquisitions.
+
+INPUT: array of hashes with acquisition parameters for each DICOM file
+
+RETURNS: hash table acquisition summary collapsed by unique acquisition
+definitions
+
+### acquisitions(self->{acqu\_Sum})
+
+Sorts the Array of Hash by acquisition number.
+
+INPUT: array of hash table acquisition summary
+
+RETURNS: acquisition listing sorted by acquisition number to be used for summary
+
+### content\_list($dcmdir)
+
+Gets DICOM info from all files in a directory.
+
+Note: I added the -k5 on August 28th 2006 because the guys in Kupio assign
```

Comment:
I would say "-k5 was added" instead of "I added" (since the author is in the Author section)